build/relsignDSCModules.sh

```diff
@@ -0,0 +1,309 @@
+                  --header "x-ms-date      : $timestamp" \
+                  --header "x-ms-blob-type : BlockBlob" \
+                  --header "x-ms-version   : $StorageVersion" \
+                  --data-binary "@${localfile}" "$url" )
+    
+    code=${output: (-3)}
+	if [ "$code" != 201 ]; then
+        echo "write_blob(): error writing $remotefile, server returned an http error code $code"
+		return 1
+    fi
+
+    return 0
+}
+
+# delete <URI> <REMOTE_FILE>
+delete_blob() {
+
+    [ ! -z "$1" ] || error "delete_blob(): expected remote URI" 1
+    [ ! -z "$2" ] || error "delete_blob(): expected remote filename" 1
+   
+    url="$1"
+    remotefile="$2"
+    
+    container=$( echo "$1" | awk -F \?  '{container=index($0,"?"); print $1}' )
+	echo "container is $container"
+    sas=$( echo "$url" | awk -F \? '{container=index($0,"?"); print $2}' )
+    url="${container}/${remotefile}?${sas}"
+    echo "url is $url"
+    timestamp=$(date +%Y-%m-%dT%H:%M:%SZ)
+    output=$(curl --silent --write-out %{http_code} --request "DELETE" \
+                  --header "x-ms-date    : $timestamp" \
+                  --header "x-ms-version : $StorageVersion" \
+                  "$url" )
+    
+    code=${output: (-3)}
+    if [[ "$code" != 202 ]]  && [[ "$code" != 404 ]]; then
+        echo "delete_blob(): error deleting $remotefile, server returned error code $code"
+		return 1
+    fi    
+    return 0
+}
+# SAS URLs
+SAS_SOURCE_URI=$1
+echo " SAS source container URL is : $1"
+SAS_DEST_URI=-$2
+echo " SAS destination container URL is : $2"
+# Build info
+BUILD_VERS=$3
+BUILD_BASEDIR=$4
+
+if [ "$BUILD_BASEDIR" = "" ]; then
+
+    echo "No build share path is specified. Default build share is mounted at: /mnt/omsdata"
+
+    BUILD_BASEDIR=/mnt/omsdata/OMSData/Builds/omsagent/develop
+
+else
+
+    echo "Build share is mounted at: $4"
+
+fi
+
+
+
+if [ ! -d $BUILD_BASEDIR ]; then
+
+    echo "Not finding Build share mounted at $BUILD_BASEDIR ..."
+
+    exit 1
+
+fi
+
+if [ "$BUILD_VERS" = "" ]; then
+
+    echo "No build version is specified."
+
+    exit 1
+
+else
+
+    echo "Build version: $3"
+
+fi
+
+BUILD_BASEDIR=${BUILD_BASEDIR}/${BUILD_VERS}
+
+
+
+if [ ! -d $BUILD_BASEDIR ]; then
+
+    echo "No build version $BUILD_VERS was found at oms build share: $BUILD_BASEDIR"
+
+    exit 1
+
+fi
+
+
+
+zip -L >/dev/null 2>&1
+
+if [ $? != 0 ]; then
+
+    echo "zip must be installed"
+
+    exit 1
+
+fi
+
+
+
+unzip -Z >/dev/null 2>&1
+
+if [ $? != 0 ]; then
+
+    echo "unzip must be installed"
+
+    exit 1
+
+fi
+
+# Working directory used for merging both x64 and x86 versions of a DSC module.
+DscModuleIntermediateDir="tmp/merging_dsc_modules"
+# Working directory where the signed package is dropped.
+DscModuleTargetDir="tmp/dsc_signed"
+# Locations of dsc modules that need to be signed
+X64Dir="$BUILD_BASEDIR/Linux_ULINUX_1.0_x64_64_Release/dsc"
+X86Dir="$BUILD_BASEDIR/Linux_ULINUX_1.0_x86_32_Release/dsc"
```

Comment:
Blech (choke, cough). There are issues that I don't fully understand here with signing, but it would be VERY good if 32-bit and 64-bit packages could be signed independently.

That would allow you to sign packages during the regular build process (when you have access to the packages from the current working directory).

The advantage here is much simpler build host setup (no need to mount the build publishing disk to re-read the builds). That's compelling, if you can possibly swing it.